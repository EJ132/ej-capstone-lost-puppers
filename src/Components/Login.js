import React, { Component } from 'react'
import '../Stylesheets/Main.css'
import Navbar from './Navbar'
import '../Stylesheets/Login.css'
import paw from '../Images/paw.png'

export default class Login extends Component {

    handleSubmitLogin = (event) => {
        event.preventDefault()
        const {user_name, password} = event.target
        const User = {user_name: user_name.value, password: password.value}
        console.log(User)

        // TokenService.saveAuthToken(
        //     TokenService.makeBasicAuthToken(User.user_name, User.password)
        //   )

        user_name.value = ''
        password.value = ''
    }

    render() {
        return (
            <div>

                <Navbar />

                <section className="loginBG">
                    <form className="login" onSubmit={this.handleSubmitLogin}>
                        <img src={paw} alt="paw"/>
                        <label>Username</label>
                        <input placeholder="puppy_lover132" name="user_name" type="user_name" required></input>
                        <label>Password</label>
                        <input placeholder="********" name="password" type="password" required></input>
                        <button type="submit">SignIn</button>
                    </form>
                </section>

                <footer>&#169; EJ Gonzalez 2019</footer>
            </div>
        )
    }
}

// line 39...44 will be filled with dogCards (lost dog cards) components