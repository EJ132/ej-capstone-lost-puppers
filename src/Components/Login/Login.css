.loginBG {
    height: 92.5vh;
    background-color: black;
    color: white;
    margin-top: 100px;
}

.login {
    background-color: black;
    width: 450px;
    height: 600px;
    margin-right: auto;
    margin-left: auto;
    margin-top: 5%;
}

.login img {
    display: block;
    margin-left: auto;
    margin-right: auto;
}

.login label, .login input, .login button {
    display: block;
    margin-left: auto;
    margin-right: auto;
    text-align: center;
}

.login label {
    padding: 10px;
}

.login input {
    width: 40%;
    border-radius: 5px;
    border: none;
    height: 6%;
}

.login button {
    margin-top: 20px;
    padding: 5px;
    width: 30%;
    height: 6%;
    border: solid 2px white;
    background-color: black;
    color: white;
    border-radius: 5px;
}

.login button:hover {
    cursor: pointer;
}

.loginBG div {
    height: 600px;
    background-color: white;
    width: 450px;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    position: relative;
    margin-top: 50px;
}

.loginBG a {
    color: white;
    position: absolute;
    bottom: 100px;
    right: 140px;
}

.loginError {
    height: 100px;
    margin-top: 100px;
    position: absolute;
    top: -110px;
    right: 23%;
}

.loginError p {
    height: 100%;
}

.regError {
    background-color: black;
    text-align: center;
}

.regError p {
    height: 100%;
}

@media screen and (max-width: 768px) {
    .loginBG {
        height: 92.5vh;
        background-color: black;
        color: white;
    }

    .loginBG div {
        height: 70%;
        width: 100%;
        background-color: black;
    }

    .loginBG a {
        color: white;
        position: absolute;
        bottom: 0;
        right: 34%;
        background-color: black;
    }
    
    .login {
        background-color: black;
        margin-left: auto;
        margin-right: auto;
        width: 100%;
        margin-top: 10%;
        position: relative;
    }
    
    .login img {
        display: block;
        margin-left: auto;
        margin-right: auto;
        width: 40%;
    }
    
    .login label, .login input, .login button {
        display: block;
        margin-left: auto;
        margin-right: auto;
        text-align: center;
    }
    
    .login label {
        padding: 10px;
    }
    
    .login input {
        width: 35%;
        max-width: 40%;
        border-radius: 5px;
        border: none;
        height: 40px;
    }
    
    .login button {
        margin-top: 20px;
        padding: 5px;
        width: 30%;
        height: 30px;
        border: 1px solid white;
        border-radius: 5px;
    
    }

}