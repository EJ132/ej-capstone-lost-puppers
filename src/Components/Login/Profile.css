@import url('https://fonts.googleapis.com/css?family=Alegreya+Sans+SC:900|Asap&display=swap');

.profile_page {
    margin-top: 40px;
    min-height: 100vh;
    height: auto;
    background-color: white;
}

.profile {
    color: white;
    padding: 40px;
    background-color: #17BEBB;
    font-family: 'Alegreya Sans SC', sans-serif;
}

.profile h2 {
    font-size: 4rem;
    text-decoration: underline;
    letter-spacing: 1px;
}

.profile h3 {
    margin-top: 10px;
    font-size: 3rem;
    margin-left: 125px;
    letter-spacing: 2px;
}

.profile p {
    padding: 10px;
    margin-top: 10px;
}

.Post {
    font-family: 'Asap', sans-serif;
    height: auto;
    color: black;
    text-align: center;
}

#yourPost {
    font-size: 3rem;
    padding: 20px 10px 0px 10px;
}


.userPosts {
    width: 70%;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    padding: 50px;
    margin-left: auto;
    margin-right: auto;
}

.addPup {
    padding: 20px;
    margin-top: 15px
}

.addPup a {
    background-color: rgb(187, 177, 235);
    padding: 10px;
    display: block;
    width: 200px;
    margin-left: auto;
    margin-right: auto;
    border-radius: 5px;
    text-decoration: none;
    color: black;
}

@media screen and (max-width: 768px) {
    
    .profile {
        color: white;
        padding: 20px;
        height: auto;
        display: flex;
        flex-direction: column;
        width: 100%;
        text-align: center;
        margin-bottom: 0px;
    }
    
    .profile h2 {
        padding: initial;
        margin-top: 20px;
        margin-bottom: 10px;
    }
    
    .profile h3 {
        margin-left: initial;
    }
    
    .profile p {
        padding: 10px;
        margin-top: 10px;
    }
    
    .Post {
        height: auto;
        color: black;
        text-align: center;
    }
    
    
    .userPosts {
        width: 100%;
        display: flex;
        flex-direction: column;
        justify-content: space-evenly;
        padding: 30px 50px 50px 50px;
        margin-left: auto;
        margin-right: auto;
    }
}