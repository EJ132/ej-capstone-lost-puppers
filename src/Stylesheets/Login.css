.loginBG {
    height: 92.5vh;
    background-color: black;
    color: white;
}

.login {
    background-color: black;
    width: 450px;
    height: 600px;
    margin-right: auto;
    margin-left: auto;
    margin-top: 5%;
}

.login img {
    display: block;
    margin-left: auto;
    margin-right: auto;
}

.login label, .login input, .login button {
    display: block;
    margin-left: auto;
    margin-right: auto;
    text-align: center;
}

.login label {
    padding: 10px;
}

.login input {
    width: 40%;
    border-radius: 5px;
    border: none;
    height: 6%;
}

.login button {
    margin-top: 20px;
    padding: 5px;
    width: 30%;
    height: 6%;
    border: none;
    border-radius: 5px;
}

.loginBG div {
    height: 600px;
    background-color: white;
    width: 450px;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    position: relative;
}

.loginBG a {
    color: white;
    position: absolute;
    bottom: 100px;
    right: 140px;
}

@media screen and (max-width: 768px) {
    .loginBG {
        height: 92.5vh;
        background-color: black;
        color: white;
    }
    
    .login {
        background-color: black;
        width: 100%;
        height: 100%;
        margin-right: auto;
        margin-left: auto;
        margin-top: 10%;
        justify-content: center;
        position: relative;
    }
    
    .login img {
        display: block;
        margin-left: auto;
        margin-right: auto;
        width: 40%;
    }
    
    .login label, .login input, .login button {
        display: block;
        margin-left: auto;
        margin-right: auto;
        text-align: center;
    }
    
    .login label {
        padding: 10px;
    }
    
    .login input {
        width: 35%;
        max-width: 40%;
        border-radius: 5px;
        border: none;
        height: 6%
    }
    
    .login button {
        margin-top: 20px;
        padding: 5px;
        width: 30%;
        height: 6%;
        border: none;
        border-radius: 5px;
    
    }

    .loginBG a {
        color: white;
        position: absolute;
        bottom: 30%;
        right: 33%;
    }

}