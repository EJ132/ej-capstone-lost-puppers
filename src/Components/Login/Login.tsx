import React, { Component } from "react";
import { Link } from "react-router-dom";

import "../Home/Home";

import config from "../../config";
import history from "../../Context/history";
import paw from "../../Images/paw.png";
import AuthApiService from "../../services/auth-api-service";
import TokenService from "../../services/token-service";
import NavBar from "../NavigationBar/NavigationBar";

import "./Login.css";

export default class Login extends Component {
  state = {
    error: null,
  };

  handleSubmitJwtAuth = (ev) => {
    ev.preventDefault();
    this.setState({ error: null });
    const { user_name, password } = ev.target;

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
      .then((res) => {
        TokenService.saveUserName(user_name.value);
        user_name.value = "";
        password.value = "";
        TokenService.saveAuthToken(res.authToken);
        fetch(`${config.API_ENDPOINT}/profile/${TokenService.getUserName()}`, {
          headers: { authorization: `bearer ${TokenService.getAuthToken()}` },
        })
          .then((res) =>
            !res.ok ? res.json().then((e) => Promise.reject(e)) : res.json()
          )
          .then((resJSON) => {
            TokenService.saveUserId(resJSON.id);
          });
        history.push("/");
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;

    return (
      <div>
        <NavBar />

        <section className="loginBG">
          <div>
            <form className="login" onSubmit={this.handleSubmitJwtAuth}>
              <section className="loginError" role="alert">
                {error && <p className="red">{error}</p>}
              </section>
              <img src={paw} alt="paw" />
              <label>Username</label>
              <input
                placeholder="puppy_lover132"
                name="user_name"
                type="user_name"
                required
              />
              <label>Password</label>
              <input
                placeholder="********"
                name="password"
                type="password"
                required
              />
              <button type="submit">SignIn</button>
            </form>
            <Link to="/register">Sign Up</Link>
          </div>
        </section>

        <footer>&#169; EJ Gonzalez 2019</footer>
      </div>
    );
  }
}
