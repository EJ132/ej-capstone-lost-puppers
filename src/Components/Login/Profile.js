import React, { Component } from 'react'
import Navbar from '../NavBar/Navbar'
import './Profile.css'
import TokenService from '../../services/token-service'
import PupApiService from '../../services/thing-api-service'
import DogTag from '../LostDogs/Dogtag'
import config from '../../config'

export default class Profile extends Component {

    state = {
        user: {
            fullname: '',
            user_name: '',
            id: null
        },
        userPost : []
    }

    //FETCHES THE CARDS AND USER

    componentDidMount() {
        PupApiService.getProfile()
            .then(resJSON => {
                this.setState({
                    user: resJSON
                })
                TokenService.saveUserId(this.state.user.id)
                this.setCards()
            })
    }

    //FILTERS ONLY TO THE CARDS NEEDED(OWNED BY USER)

    setCards = () => {

        PupApiService.getpups()
        .then(resJSON => {
            // eslint-disable-next-line eqeqeq
            const userPups = resJSON.filter(dog => {

                if(dog.owner === this.state.user.id){
                        return dog
                    }

                // eslint-disable-next-line array-callback-return
                return;
                })
            this.setState({
                userPost: userPups
            })
        })
    }

    
    render(){
        const {user, userPost} = this.state
        return (
            <div>
                <Navbar />
                <div className="profile_page">
                  <div className="profile">
                        <h2>WELCOME</h2>
                        <h3>{user.fullname}</h3>
                    </div>

                    <div className="Post">
                    <h2 id="yourPost">Your Posts ({userPost.length})</h2>
                        <div className="userPosts">
                        {userPost.map(dogCard => {
                            return <DogTag 
                                name={dogCard.name}
                                img={dogCard.image}
                                description={dogCard.description}
                                category={dogCard.category}
                                dateCreated={dogCard.date_created}
                                id={dogCard.id}
                                key={dogCard.id}
                                owner={dogCard.owner}
                                />
                        })}
                        </div>
                    </div>
                </div>

                <footer>&#169; EJ Gonzalez 2019</footer>
            </div>
        )
    }
}