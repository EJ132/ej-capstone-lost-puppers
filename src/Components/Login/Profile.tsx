import React, { Component } from "react";

import PupApiService from "../../services/thing-api-service";
import TokenService from "../../services/token-service";
import DogTag from "../LostDogs/Dogtag";
import NavigationBar from "../NavigationBar/NavigationBar";

import "./Profile.css";

export default class Profile extends Component {
  state = {
    user: {
      fullname: "",
      user_name: "",
      id: null,
    },
    userPost: [],
  };

  // FETCHES THE CARDS AND USER

  componentDidMount() {
    PupApiService.getProfile().then((resJSON) => {
      this.setState({
        user: resJSON,
      });
      TokenService.saveUserId(this.state.user.id);
      this.setCards();
    });
  }

  // FILTERS ONLY TO THE CARDS NEEDED(OWNED BY USER)

  setCards = () => {
    PupApiService.getpups().then((resJSON) => {
      // eslint-disable-next-line eqeqeq
      const userPups = resJSON.filter((dog) => {
        if (dog.owner === this.state.user.id) {
          return dog;
        }

        // eslint-disable-next-line array-callback-return
      });
      this.setState({
        userPost: userPups,
      });
    });
  };

  render() {
    const { user, userPost } = this.state;
    return (
      <div>
        <NavigationBar />
        <div className="profile_page">
          <div className="profile">
            <h2>WELCOME</h2>
            <h3>{user.fullname}</h3>
          </div>

          <div className="Post">
            <h2 id="yourPost">Your Posts ({userPost.length})</h2>
            <div className="userPosts">
              {userPost.map((dogCard) => {
                return (
                  <DogTag
                    name={dogCard.name}
                    img={dogCard.image}
                    description={dogCard.description}
                    category={dogCard.category}
                    dateCreated={dogCard.date_created}
                    id={dogCard.id}
                    key={dogCard.id}
                    owner={dogCard.owner}
                  />
                );
              })}
            </div>
          </div>
        </div>

        <footer>&#169; EJ Gonzalez 2019</footer>
      </div>
    );
  }
}
