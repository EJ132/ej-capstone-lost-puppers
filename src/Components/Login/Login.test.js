import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Login from './Login'

describe(`Login component`, () => {

  it('renders a Login page by default', () => {
    const wrapper = shallow(<Login />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})
