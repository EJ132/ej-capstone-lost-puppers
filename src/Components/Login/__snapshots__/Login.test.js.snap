// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders a Login page by default 1`] = `
<div>
  <Navbar />
  <section
    className="loginBG"
  >
    <div>
      <form
        className="login"
        onSubmit={[Function]}
      >
        <section
          className="loginError"
          role="alert"
        />
        <img
          alt="paw"
          src="paw.png"
        />
        <label>
          Username
        </label>
        <input
          name="user_name"
          placeholder="puppy_lover132"
          required={true}
          type="user_name"
        />
        <label>
          Password
        </label>
        <input
          name="password"
          placeholder="********"
          required={true}
          type="password"
        />
        <button
          type="submit"
        >
          SignIn
        </button>
      </form>
      <Link
        to="/register"
      >
        Sign Up
      </Link>
    </div>
  </section>
  <footer>
    Â© EJ Gonzalez 2019
  </footer>
</div>
`;
